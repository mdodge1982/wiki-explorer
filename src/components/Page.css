.Page {
	margin: 0;
}
.Page h5 {
	margin: 0;
	padding: .4em 1em;
	border-bottom: 1px solid #ddd;
	cursor: pointer;
}
