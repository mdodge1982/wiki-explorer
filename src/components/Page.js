import React, {Component} from 'react';
import {connect} from 'react-redux';
import {selectPage} from '../actions';
import './Page.css';

class PageComonent extends Component {
	render() {
		return (
			<li className="Page"><h5 onClick={() => this.handleClick()}>{this.props.page.title}</h5></li>
		);
	}
	handleClick() {
		this.props.dispatch(selectPage(this.props.page));
	}
}

const Page = connect()(PageComonent);

export default Page;
