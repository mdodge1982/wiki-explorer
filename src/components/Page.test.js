import React from 'react';
import Page from './Page';
import {Provider} from 'react-redux';
import store from '../store';
import renderer from 'react-test-renderer';

jest.mock('../actions/websocket');

test('Test render', () => {
	const page = {
		title: 'Test Title'
	}
	const component = renderer.create(
		<Provider store={store}>
			<Page page={page}/>
		</Provider>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();

});
