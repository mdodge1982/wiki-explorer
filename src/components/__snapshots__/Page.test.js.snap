// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render 1`] = `
<li
  className="Page"
>
  <h5
    onClick={[Function]}
  >
    Test Title
  </h5>
</li>
`;
